kafka-producer-java/src/main/java/Producer.java

```diff
@@ -27,15 +29,8 @@ public boolean ready() {
     public boolean produce(ProducerRequest producerRequest) {
```

Comment:
it makes more sense to create an abstract producer, with two classes blockingProducer, and AsyncProducer, each one of them implements the producer differently